common/test/src/TestUtils.h

```diff
@@ -122,4 +123,55 @@ namespace TrenchBroom {
     };
 
     GlobMatcher MatchesGlob(const std::string& glob);
+
+    template <typename T, std::size_t S>
+    class UnorderedApproxVecMatcher : public Catch::MatcherBase<std::vector<vm::vec<T,S>>> {
+    private:
+        std::vector<vm::vec<T,S>> m_expected;
+        T m_epsilon;
+    public:
+        explicit UnorderedApproxVecMatcher(const std::vector<vm::vec<T,S>>& expected, const T epsilon) :
+            m_expected(expected),
+            m_epsilon(epsilon) {}
+
+        bool match(const std::vector<vm::vec<T,S>>& actual) const override {
+            if (actual.size() != m_expected.size()) {
+                return false;
+            }
+
+            for (auto& actualElement : actual) {
```

Comment:
You could use `std::is_permutation` with a custom predicate here.

Reply:
According to https://en.cppreference.com/w/cpp/algorithm/is_permutation it's undefined behaviour if the predicate isn't transitive, and ours isn't.

Reply:
Could use std::any_of instead of the inner for loop if you want.. (need to `#include <algorithm>` though)

Reply:
Oh damn, I keep forgetting this. Sorry about that.

Reply:
No need for std::any. That doesn't make it clearer either IMO.